### PR TITLE
Upload ipython notebook files

### PR DESCRIPTION
**Pull request by malev**
- Upload ipython notebook files
- Upload signature as metadata
- Since ipython notebook files lack on version, I'm using  `1.0` as a default version

@srossross, would you mind taking a look?

### CODE REVIEW

File: binstar_client/inspect_package/ipynb.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import json
+
+
+class IPythonNotebook(object):
+    def __init__(self, filename, fileobj):
+        content = json.loads(fileobj.read())
+        self.name = content['metadata']['name'] or filename
+        self.signature = content['metadata']['signature']
+
+        if 'version' in content['metadata']:
```
Comment (srossross) [inline binstar_client/inspect_package/ipynb.py]: Why not use: 

```
self.version = content['metadata'].get('version', '1.0')
```

File: binstar_client/inspect_package/ipynb.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    package_data = {
+        'name': ipython_notebook.name,
+        'summary': 'IPython notebook'
+    }
+    release_data = {
+        'version': ipython_notebook.version,
+        'description': ''
+    }
+    file_data = {
+        'basename': ipython_notebook.name,
```
Comment (srossross) [inline binstar_client/inspect_package/ipynb.py]: basename should be `os.path.basename(filename)`

Comment (malev) [inline binstar_client/inspect_package/ipynb.py]: done

**Comment by srossross** [comment]
@malev can you add some tests? The code looks good.

**Comment by malev** [comment]
@srossross tested!

File: binstar_client/inspect_package/ipynb.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import json
+import os
+
+
+class IPythonNotebook(object):
+    def __init__(self, filename, fileobj):
+        content = json.loads(fileobj.read())
+        self.name = content['metadata']['name'] or filename
```
Comment (srossross) [inline binstar_client/inspect_package/ipynb.py]: This needs to be 
`self.name = content['metadata']['name'] or self.basename`

Package names can not contain `/`

Comment (srossross) [inline binstar_client/inspect_package/ipynb.py]: You probably can strip the '.ipynb' file extension as well.

Comment (malev) [inline binstar_client/inspect_package/ipynb.py]: done!

**Pull request merged**